### PR TITLE
test(compatibility-suite): Implement V3 scenarios

### PR DESCRIPTION
**Pull request by tienvx**
* Add services
  * Body validation
    * `BodyStorage`
    * `BodyValidator`
  * Convert fixture file to generators
    * `GeneratorParser`
    * `GeneratorConverter`
  * Apply generators to http request
    * `RequestGeneratorBuilder`
    * `ResponseGeneratorBuilder`
  * Apply generators to message
    * `MessageGeneratorBuilder`
  * Starting a server to capture generated value
    * `GeneratorServer`
  * Writing message to json pact file
    * `MessagePactWriter`
* Implement v3 scenarios
  * http
    * consumer
    * provider
  * message
    * consumer
    * provider
  * generators
  * matching rules
* Register V3 suites to behat
* Add v3 job to Compatibility Suite workflow
* There are 2 failed scenarios in v3 suite. Waiting for these PRs:
  * https://github.com/pact-foundation/pact-compatibility-suite/pull/6
  * https://github.com/pact-foundation/pact-compatibility-suite/pull/5

### CODE REVIEW

**Comment by Lewiscowles1986** [comment]
Can you force the failing cases to skip or xfail for now?

**Comment by tienvx** [comment]
Updated using tag filtering

**Comment by tienvx** [comment]
~~Updated using tag filtering~~

nevermind, it doesn't work as expected, close for now while waiting for the solution

**Comment by tienvx** [comment]
Fixed. It's a bit complicated than I thought.

There is no way I can set this filter to `v3_message_provider` suite's config:

```yaml
            filters:
                name: '/^((?!Kafka|binary body \(negative).)*$/'
```

The `filters` array is flipped, probably because of this code https://github.com/Behat/Behat/blob/master/src/Behat/Behat/Gherkin/ServiceContainer/GherkinExtension.php#L97.

There is another option is to use `--name` flag. But because of the way it work (scenarios are not filtered if feature is matched), I have to filter out the feature as well.

```
 --name '/^((?!Kafka|binary body \(negative|Message provider).)*$/'
```

**Review by Lewiscowles1986 — APPROVED** [review]
Hard to argue with well-documented, working code. Hope the other PR's land safely

File: compatibility-suite/tests/Service/MessagePactWriterInterface.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace PhpPactTest\CompatibilitySuite\Service;
+
+use PhpPact\Config\PactConfigInterface;
+
+interface MessagePactWriterInterface
+{
+    public function write(string $name, string $body, string $consumer = 'c', string $provider = 'p', string $mode = PactConfigInterface::MODE_OVERWRITE): void;
```
Comment (Lewiscowles1986) [inline compatibility-suite/tests/Service/MessagePactWriterInterface.php:9]: guessing the constants here are outside of control to make enums

(consumer = c and provider = p)

Comment (tienvx) [reply compatibility-suite/tests/Service/MessagePactWriterInterface.php:9]: This code look bad. I will try to improve by creating a DTO in a future PR.

Comment (tienvx) [reply compatibility-suite/tests/Service/MessagePactWriterInterface.php:9]: Create a PR to fix this hard coding https://github.com/pact-foundation/pact-php/pull/433

**Pull request merged**